src/components/entity/state-badge.ts

```diff
@@ -55,8 +56,11 @@ class StateBadge extends LitElement {
     };
     if (stateObj) {
       // hide icon if we have entity picture
-      if (stateObj.attributes.entity_picture && !this.overrideIcon) {
-        let imageUrl = stateObj.attributes.entity_picture;
+      if (
+        (stateObj.attributes.entity_picture && !this.overrideIcon) ||
+        this.overrideImage
+      ) {
+        let imageUrl = stateObj.attributes.entity_picture || this.overrideImage;
```

Comment:
If overrideIcon is set to `false` and `overrideImage` is provided, `entity_picture` is used now . I think you need to reverse this line

```suggestion
        let imageUrl = this.overrideImage || stateObj.attributes.entity_picture;
```